### PR TITLE
The asteroid-with-engines shuttle now announces people it runs over

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: When the asteroid-with-engines shuttle runs people over, there is
an anouncement with their names, allowing you to learn from their
mistakes.
/:cl:

Gotta get your name on the board.

### CODE REVIEW

**Comment by chesse20** [comment]
add this please and give me one

**Comment by RandomMarine** [comment]
\>be geneticist
\>create army of monkeybrain men
\>stuff them all in lockers and bodybags
\>drag them into range of the shuttle
\>shuttle docks
\>instant cascade of announcements floods across the chat

**Comment by darkpaladin109** [comment]
Yeah, you should make it so that NPC's don't count for that, if possible.

File: code/datums/components/roadkill_counter.dm [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/datum/component/roadkill_counter
+	var/list/names
+
+/datum/component/roadkill_counter/Initialize()
+	if(!istype(parent, /obj/docking_port/mobile))
+		. = COMPONENT_INCOMPATIBLE
+		CRASH("A roadkill counter component has been applied to an incorrect object. parent: [parent]")
+
+	names = list()
```
Comment (Cyberboss) [inline code/datums/components/roadkill_counter.dm]: make this a lazy list

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by chaoko99** [comment]
Reminds me of when slimepeople could gain stacks from the toxins they'd take from their own blood. They had to use a shuttle to stop me.

**Comment by optimumtact** [comment]
stale ping for reopen etc, just needs that lazy list

**Pull request closed without merging**